cmd/notation/verify.go

```diff
@@ -155,13 +157,29 @@ func runVerify(command *cobra.Command, opts *verifyOpts) error {
 
 func checkVerificationFailure(outcomes []*notation.VerificationOutcome, printOut string, err error) error {
 	// write out on failure
-	if err != nil || len(outcomes) == 0 {
-		if err != nil {
-			var errorVerificationFailed notation.ErrorVerificationFailed
-			if !errors.As(err, &errorVerificationFailed) {
-				return fmt.Errorf("signature verification failed: %w", err)
+	if err != nil {
+		// reference: https://pkg.go.dev/errors#Join
+		if joinedError, ok := err.(interface{ Unwrap() []error }); ok {
+			errArray := joinedError.Unwrap()
+			// if err is a joined error, then it always starts with a general
+			// error message followed by indivisual notation.VerificationOutcome
+			// errors.
+			for _, outcomeError := range errArray[1:] {
```

Comment:
what if `len(errArray) == 0`?

Reply:
Although this should never happen because the logic is guarded by notation-go, I'll add an extra check just in case.